src/buffer/out/textBuffer.cpp

```diff
@@ -4,15 +4,80 @@
 #include "precomp.h"
 
 #include "textBuffer.hpp"
-#include "CharRow.hpp"
+
+#include <til/hash.h>
 
 #include "../renderer/base/renderer.hpp"
 #include "../types/inc/utils.hpp"
 #include "../types/inc/convert.hpp"
 #include "../../types/inc/Utf16Parser.hpp"
 #include "../../types/inc/GlyphWidth.hpp"
 
-#pragma hdrstop
+struct BufferAllocator

```

Comment:
I'll chat with you more about this offline, but I'm having trouble understanding the value of the `BufferAllocator`.

Regardless, I think it'd be valuable to add a comment over it introducing its purpose.

Reply:
`BufferAllocator` represents the dark side of my shobby coding practices. 
1. I want to allocate the backing storage for the entire `TextBuffer` all at once, so I'm calling `VirtualAlloc` with how much memory I need.
2. I want to split the allocated memory up into small row-sized chunks and thus need an iterator or something.
3. It should be minimally invasive so that the diff is small.

This struct represents the union of those 3 of my wishes. It's an allocator-iterator-holder-thingy basically. It computes how much memory I need and calls `VirtualAlloc` (area of responsibility 1) and then also allows me to split up the memory into chunks (area of responsibility 2). This should technically be two seperate pieces of code, but I feel like the struct is compact enough that anyone could rewrite it from scratch within a reasonable amount of time.